xbmc/video/ContextMenus.cpp

```diff
@@ -119,6 +122,39 @@ bool CVideoMarkUnWatched::Execute(const std::shared_ptr<CFileItem>& item) const
   return true;
 }
 
+bool CVideoBrowse::IsVisible(const CFileItem& item) const
+{
+  if (item.IsFileFolder(EFILEFOLDER_MASK_ONBROWSE))
```

Comment:
Is there any other way to avoid special casing here? Context menus are common code that ideally shouldn't need to know anything about the calling context. I mean if you somehow could blacklist specific items on advance in the window you could avoid the condition here

Reply:
Can be done, but sensible code areas I have zero knowledge (`CGUIMediaWindow` and what not) needs to be touched for this (at least this is my understanding). Something for another day, I'd say.